src/type.h

```diff
@@ -205,9 +205,10 @@ class Type : public Traceable {
         not to clash. */
     virtual std::string Mangle() const = 0;
 
-    /** Returns a string that is the declaration of the same type in C
-        syntax. */
-    virtual std::string GetCDeclaration(const std::string &name) const = 0;
+    /** Returns a string that is the declaration of the same type in C or C++
+        syntax. The second argument controls which syntax C (false) or
+        C++ (true) is generated. */
+    virtual std::string GetDeclaration(const std::string &name, bool cpp = false) const = 0;
```

Comment:
It's up to you, but I would prefer to have 
```
enum DeclarationType {
C,
Cpp,
}
```
and then use it instead of bool variable.
I think it would improve readability of the code and I prefer to avoid default parameters when possible because it's a potential source of future errors.

Reply:
I totally agree with Arina, please do this change.

Reply:
I have introduced enum and removed default parameter.